API/Common.pm

```diff
@@ -107,13 +107,25 @@ sub _precacheAlbum {
 		$album->{genre} = $album->{genre}->{name};
 		$album->{image} = __PACKAGE__->getImageFromImagesHash($album->{image}) || '';
 
+		# If the user pref is for classical music enhancements to the display, is this a classical release or has the user added the genre to their custom classical list?
+		$isClassique = 0;
+		if ( $prefs->get('useClassicalEnhancements') ) {
+			my %genreList = map { $_ => 1 } split '\s*,\s*', $prefs->get('classicalGenres');	
+			if ( $album->{genres_list} && grep(/Classique/,@{$album->{genres_list}}) || $genreList{$album->{genre}} ) {
+			$isClassique = 1;
+			}
+		}
+
 		my $albumInfo = {
 			title  => $album->{title},
 			id     => $album->{id},
 			artist => $album->{artist},
 			image  => $album->{image},
 			year   => (localtime($album->{released_at}))[5] + 1900,
 			goodies=> $album->{goodies},
+			genre  => $album->{genre},
+			genres_list => $album->{genres_list},
+			isClassique => $isClassique,
```

Comment:
Wouldn't that fail to pick up changes in the `classicalGenres` list? You might need to evaluate that flag whenever you read the data.

Reply:
Seems to work fine as it is, both before and after implementing the initGenreMap subroutine you suggested.

Reply:
But if you set `isClassique` to false, then changed `classicalGenres` to include this track's genre after it has been cached, `isClassique` would not be updated to be truthy, would it?

Reply:
From my reading of the code (remember Perl is a foreign language to me), it's only the raw results from Qobuz that get cached, therefore `isClassique` is re-evaluated every time - 

```
Plugin.pm
   QobuzGetTracks
      getAlbum (API.pm)
         _get (album data from cache or Qobuz)
            _precacheAlbum (Common.pm)
               evaluates $isClassique based on Qobuz/cache results & user prefs
               sets isClassique in $albumInfo map
               _precacheTracks
                  _precacheTrack  
                     sets isClassique in $meta map
```

I could be horribly wrong, of course, but it explains the behaviour I am seeing.

Reply:
...and then `$meta` is stored in the cache, including the `$isClassique` flag, isn't it? See Line 139 (https://github.com/pierrepoulpe/SqueezeboxQobuz/pull/32/files/3d0073e0dc6df56651b6e2f38271314103572573#diff-03c728c5508826c89b0befa221adcc73ac401f2c9dd3444bf9d27eb03586f173R193)

Reply:
OK, I think I understand what you are saying: future functionality might use the classique flag straight out of the cache and end up using a stale one. Correct?

So what do we need to do? When the user pref  genre list changes, retrieve the cache, update the flag, then write the cache back?

Reply:
It's not future functionality - see https://github.com/pierrepoulpe/SqueezeboxQobuz/blob/master/API.pm#L511. We don't write to the cache for nothing. We use it to speed up future lookups (eg. when fetching metadata to be displayed in the UI).

I think instead of storing the flag itself you should store all the data needed to evaluate the value at run time. Then whenever you get track information, no matter whether freshly fetched from remote or out of the cache, you evaluate the flag once again.

Reply:
I must admit I'm getting a bit confused here (I often do!).  Isn't that effectively what I'm doing already? I get the Qobuz genre_list tag from their metadata (cached or not) and the additional genre list from the user Prefs, then set the isClassique flag from those data. This happens every time the browse album list or the playlist changes.

Perhaps the issue is that the isClassique flag doesn't really belong in the metadata maps, but adding it there was a very easy way of making it available in the places where it needs to be checked.

Reply:
In ProtocolHandler you potentially get the data from the cache, incl. that flag. In that case you don't evaluate the genres any more but rely on the `isClassique` flag coming from the cache. It's probably the only place where you're not dealing with fresh data. 

We could consider this negligible. That method is called while a track is playing. If you change the genre list in your settings while an affected track is playing... oh well. 

Let's consider this good enough.

Reply:
The behaviour I have seen during testing is that the playing item in the playlist gets updated if you change the preference to include/exclude the genre, then refresh the album listing. Then the whole playlist gets updated on any addition to the playlist, track change, or user initiated seek or skip event. This is using the Material skin, not sure if that makes a difference. 

I don't mind getting to the bottom of this (or just implementing any further changes you suggest). It's a great learning exercise. But equally, if you're happy, I'm happy, and I'll update GitHub with things as they are.

Which do you prefer?

Reply:
I think we can leave it as it is and come back should there be an issue.